roachtest cover test

RaduBerinde opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110920)
<!-- Reviewable:end -->

RaduBerinde committed:
coverage: add roachtest nightly scripts

Epic: none
Release note: None

RaduBerinde committed:
roachtest: add support for collecting Go coverage

PR #110280 added instrumentation to the cockroach binary. When built
in this mode, cockroach dumps coverage counters to the directory
specified in `BAZEL_COVER_DIR` env var. Note that this is similar to
Go's recent support for binary instrumentation
(https://go.dev/testing/coverage/); we will eventually switch to using
that when Bazel supports building in that mode.

This commit adds a `-go-cover` flag to `roachtest`. When this flag is
set, the `BAZEL_COVER_DIR` env var is set and after the test completes
we copy the contents to the artifacts (similar to the `perf`
artifacts).

Epic: none
Release note: None
